### PR TITLE
PySpark Test

### PR DESCRIPTION
**Pull request by ifilonenko**
Added test for checking python bindings

### CODE REVIEW

**Comment by holdenk** [comment]
I'm super happy we have integration tests, I'm super sad they aren't part of the main Spark repo. Is there anything I can do to help?

**Comment by ssuchter** [comment]
Yes… There is a PR for that, but I kind of lost steam on it (had to do
other work things) on the very last bit - integrating the calls into
run_tests.py. If you are familiar with that script and could suggest
where/how they should be called from that, it’d help.

See https://github.com/apache/spark/pull/20697

Thanks,
Sean

On April 20, 2018 at 1:15:16 PM, Holden Karau (notifications@github.com)
wrote:

I'm super happy we have integration tests, I'm super sad they aren't part
of the main Spark repo. Is there anything I can do to help?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/apache-spark-on-k8s/spark-integration/pull/46#issuecomment-383210029>,
or mute the thread

**Comment by holdenk** [comment]
Hey @ifilonenko really like this start of an integration test, would love to see something using `--py-files` as well in the integration side given how difficult it was for us to get even the basics of that working together.

**Comment by ifilonenko** [comment]
please retest this

File: src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala [hunk 1]
```diff
@@ -184,20 +186,72 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
...
+      executorPodChecker = doBasicExecutorPyPodCheck,
+      appLocator = appLocator,
+      isJVM = false,
+      pyFiles = Some(PYSPARK_CONTAINER_TESTS))
+  }
+
+  test("Run PySpark with Python3 to test a pyfiles example") {
+    sparkAppConf
+      .set("spark.kubernetes.container.image", s"${getTestImageRepo}/spark-py:${getTestImageTag}")
+      .set("spark.kubernetes.pyspark.pythonversion", "3")
```
Comment (liyinan926) [inline src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala:224]: Nit: looks like `pythonVersion` is the right one based on the style of existing Spark config properties.

Comment (ifilonenko) [reply src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala:224]: hehe... well... I will make sure to change that style in a follow-up PR to core.

Comment (liyinan926) [reply src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala:224]: Sounds good.

**Comment by ifilonenko** [comment]
Same as last PR (because of the introduction of new files into `examples/`), need to wait until a new distribution is made by the daily `make-distribution-builder` seen here: https://amplab.cs.berkeley.edu/jenkins/job/testing-k8s-make-spark-distribution-master/128/

**Pull request closed without merging**